jina/types/document/graph.py

```diff
@@ -0,0 +1,336 @@
+from typing import Optional, Iterator, Tuple, Dict, Iterable
+
+import numpy as np
+
+from . import Document, DocumentSourceType
+from ..arrays import ChunkArray
+from ..struct import StructView
+from ..ndarray.sparse.scipy import SparseNdArray
+from ...importer import ImportExtensions
+from ...logging.predefined import default_logger
+
+__all__ = ["GraphDocument"]
+
+if False:
+    from scipy.sparse import coo_matrix
+
+
+class GraphDocument(Document):
+    """
+    :class:`GraphDocument` is a data type created based on Jina primitive data type :class:`Document`.
+
+    It adds functionality that lets you work with a `Document` as a `graph` where all its chunks are the nodes in the `graph`.
```

Comment:
Maybe we could add to the documentation that it is a `directed graph`.

Reply:
makes sense @DavidSanwald

Reply:
Fully Agree